containers/unit_tests/TestOffsetView.hpp

```diff
@@ -57,11 +57,15 @@ void test_offsetview_construction() {
 
   ASSERT_EQ("firstOV", ov.label());
 
+#ifdef KOKKOS_ENABLE_DEPRECATION_WARNINGS
```

Comment:
Question (and not a blocker):  why not put the `KOKKOS_ENABLE_DEPRECATION_WARNINGS` check directly in the `KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH` and `KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_POP` macros?

Reply:
Because we might want to disable deprecated warnings not emitted by us and because then the macro does what it says.

Reply:
Is this the use-case for the variety of macros you have in mind? 

```c++
#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4 
#ifdef KOKKOS_ENABLE_DEPRECATION_WARNINGS //expand KOKKOS_DEPRECATED macro ([[deprecated]])
using old_type KOKKOS_DEPRECATED_WITH_COMMENT(
      "Use new type") = int; //Compiler emits warning
KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()  //Selectively disable warning just for this
using another_old_type KOKKOS_DEPRECATED_WITH_COMMENT(
      "Use another new type") = int;
KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_POP()
#endif
#endif
```

Reply:
:(